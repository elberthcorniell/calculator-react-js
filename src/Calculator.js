import { Component } from 'react';
import { ButtonPanel, Display } from './components';
import { calculate } from './logic';
import './App.css';

class Calculator extends Component {
  constructor() {
    super();
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const newState = calculate(this.state, buttonName);
    this.setState(newState);
  }

  render() {
    const { total } = this.state;
    return (
      <div className="calculator">
        <Display result={total} />
        <ButtonPanel clickHandler={this.handleClick} />
      </div>
    );
  }
}

export default Calculator;
