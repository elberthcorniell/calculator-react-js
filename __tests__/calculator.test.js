import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../src/Calculator';

describe('test', () => {
  test('Render as spected', () => {
    render(<Calculator />);
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  test('Loads and displays button 0', () => {
    render(<Calculator />);
    expect(screen.getAllByText('0')).toBeDefined();
  });

  test('Loads and displays button 1', () => {
    render(<Calculator />);
    expect(screen.getAllByText('1')).toBeDefined();
  });

  test('Loads and displays button 2', () => {
    render(<Calculator />);
    expect(screen.getAllByText('2')).toBeDefined();
  });

  test('Loads and displays button 3', () => {
    render(<Calculator />);
    expect(screen.getAllByText('3')).toBeDefined();
  });

  test('Loads and displays button 4', () => {
    render(<Calculator />);
    expect(screen.getAllByText('4')).toBeDefined();
  });

  test('Loads and displays button 5', () => {
    render(<Calculator />);
    expect(screen.getAllByText('5')).toBeDefined();
  });

  test('Loads and displays button 6', () => {
    render(<Calculator />);
    expect(screen.getAllByText('6')).toBeDefined();
  });

  test('Loads and displays button 7', () => {
    render(<Calculator />);
    expect(screen.getAllByText('7')).toBeDefined();
  });

  test('Loads and displays button 8', () => {
    render(<Calculator />);
    expect(screen.getAllByText('8')).toBeDefined();
  });

  test('Loads and displays button 9', () => {
    render(<Calculator />);
    expect(screen.getAllByText('9')).toBeDefined();
  });

  test('Loads and displays button -', () => {
    render(<Calculator />);
    expect(screen.getAllByText('-')).toBeDefined();
  });

  test('Loads and displays button +', () => {
    render(<Calculator />);
    expect(screen.getAllByText('+')).toBeDefined();
  });

  test('Loads and displays button x', () => {
    render(<Calculator />);
    expect(screen.getAllByText('x')).toBeDefined();
  });

  test('Loads and displays button รท', () => {
    render(<Calculator />);
    expect(screen.getAllByText('รท')).toBeDefined();
  });

  test('Loads and displays button AC', () => {
    render(<Calculator />);
    expect(screen.getAllByText('AC')).toBeDefined();
  });

  test('Loads and displays button =', () => {
    render(<Calculator />);
    expect(screen.getAllByText('=')).toBeDefined();
  });
});
