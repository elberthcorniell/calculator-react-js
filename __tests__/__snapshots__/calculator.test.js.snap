// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Render as spected 1`] = `"<div><div class=\\"calculator\\"><div class=\\"display\\"><strong>0</strong></div><div class=\\"button-grid\\"><button type=\\"button\\" class=\\"button-calc \\">AC</button><button type=\\"button\\" class=\\"button-calc \\">+/-</button><button type=\\"button\\" class=\\"button-calc \\">%</button><button type=\\"button\\" class=\\"button-calc orange\\">รท</button></div><div class=\\"button-grid\\"><button type=\\"button\\" class=\\"button-calc \\">7</button><button type=\\"button\\" class=\\"button-calc \\">8</button><button type=\\"button\\" class=\\"button-calc \\">9</button><button type=\\"button\\" class=\\"button-calc orange\\">x</button></div><div class=\\"button-grid\\"><button type=\\"button\\" class=\\"button-calc \\">4</button><button type=\\"button\\" class=\\"button-calc \\">5</button><button type=\\"button\\" class=\\"button-calc \\">6</button><button type=\\"button\\" class=\\"button-calc orange\\">-</button></div><div class=\\"button-grid\\"><button type=\\"button\\" class=\\"button-calc \\">1</button><button type=\\"button\\" class=\\"button-calc \\">2</button><button type=\\"button\\" class=\\"button-calc \\">3</button><button type=\\"button\\" class=\\"button-calc orange\\">+</button></div><div class=\\"button-grid zero\\"><button type=\\"button\\" class=\\"button-calc \\">0</button><button type=\\"button\\" class=\\"button-calc \\">.</button><button type=\\"button\\" class=\\"button-calc orange\\">=</button></div></div></div>"`;
